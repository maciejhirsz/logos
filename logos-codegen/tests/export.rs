#![cfg(feature = "debug")]

use std::path::Path;

use insta::assert_snapshot;

#[test]
fn test_complex() {
    test_export("complex");
}

fn test_export(case: &str) {
    let mut input_path = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/export");
    input_path.push(case);
    input_path.push("input.rs");

    let input = std::fs::read_to_string(input_path).expect("Unable to read input.rs");

    let _ = logos_codegen::generate(input.parse().expect("Unable to parse input.rs"));

    let generated_dot = std::fs::read_to_string(format!("{case}_export_tmp/{case}.dot")).expect("Unable to read dot file");
    let generated_mermaid = std::fs::read_to_string(format!("{case}_export_tmp/{case}.mmd")).expect("Unable to read mermaid file");

    assert_snapshot!(generated_dot);
    assert_snapshot!(generated_mermaid);

    // cleanup
    let _ = std::fs::remove_dir_all(format!("{case}_export_tmp"));
}
