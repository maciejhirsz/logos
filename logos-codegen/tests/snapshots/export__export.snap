---
source: logos-codegen/tests/export.rs
expression: generated_dot
---
digraph {
node[shape=box];
splines=ortho;
n1024ctx0[label="State 1024 (ctx 0)",color=green];
n1536ctx0[label="State 1536 (ctx 0)",color=green];
n1536ctx0->n7680ctx0 [label="r"];
n2048ctx2[label="State 2048 (ctx 2)",color=green];
n2560ctx2[label="State 2560 (ctx 2)",color=green];
n2560ctx2->n3584ctx3 [label="\\x00..=`|{..=\\xff"];
n2560ctx2->n4096ctx3 [label="a..=z"];
n2560ctx2->n3584ctx3 [label="EOI"];
n3072ctx2[label="State 3072 (ctx 2)",color=green];
n3072ctx2->n3584ctx3 [label="\\x00..=`|{..=\\xff"];
n3072ctx2->n4096ctx3 [label="a..=b|d..=z"];
n3072ctx2->n4608ctx3 [label="c"];
n3072ctx2->n3584ctx3 [label="EOI"];
n3584ctx3[label="State 3584 (ctx 3)",color=green];
n4096ctx3[label="State 4096 (ctx 3)",color=green];
n4096ctx3->n3584ctx3 [label="\\x00..=`|{..=\\xff"];
n4096ctx3->n4096ctx3 [label="a..=z"];
n4096ctx3->n3584ctx3 [label="EOI"];
n4608ctx3[label="State 4608 (ctx 3)",color=green];
n4608ctx3->n3584ctx3 [label="\\x00..=`|{..=\\xff"];
n4608ctx3->n4096ctx3 [label="a..=s|u..=z"];
n4608ctx3->n5120ctx3 [label="t"];
n4608ctx3->n3584ctx3 [label="EOI"];
n5120ctx3[label="State 5120 (ctx 3)",color=green];
n5120ctx3->n5632ctx1 [label="\\x00..=`|{..=\\xff"];
n5120ctx3->n6144ctx1 [label="a..=z"];
n5120ctx3->n5632ctx1 [label="EOI"];
n5632ctx1[label="State 5632 (ctx 1)",color=green];
n6144ctx1[label="State 6144 (ctx 1)",color=green];
n6144ctx1->n3584ctx3 [label="\\x00..=`|{..=\\xff"];
n6144ctx1->n4096ctx3 [label="a..=z"];
n6144ctx1->n3584ctx3 [label="EOI"];
n6656[label="State 6656",color=black];
n6656->n7168 [label="a..=r|t..=z"];
n6656->n8192 [label="s"];
n7168[label="State 7168",color=black];
n7168->n1024ctx0 [label="\\x00..=\\xff"];
n7168->n1024ctx0 [label="EOI"];
n7680ctx0[label="State 7680 (ctx 0)",color=black];
n7680ctx0->n2048ctx2 [label="\\x00..=`|{..=\\xff"];
n7680ctx0->n2560ctx2 [label="a..=t|v..=z"];
n7680ctx0->n3072ctx2 [label="u"];
n7680ctx0->n2048ctx2 [label="EOI"];
n8192[label="State 8192",color=black];
n8192->n1024ctx0 [label="\\x00..=s|u..=\\xff"];
n8192->n1536ctx0 [label="t"];
n8192->n1024ctx0 [label="EOI"];
}
